### DISCUSSION TITLE
Warning: detected hash type "ZIP", but the string is also recognized as "ZIP-opencl"

### DISCUSSION THREAD

**Discussion by Theletha** [illsk1lls/ZipRipper#43] [Q&A] [answered]
What do i do with this error? 

```
Running JohnTheRipper...

Warning: detected hash type "ZIP", but the string is also recognized as "ZIP-opencl"
Use the "--format=ZIP-opencl" option to force loading these as that type instead
Using default input encoding: UTF-8
Loaded 19 password hashes with 19 different salts (ZIP, WinZip [PBKDF2-SHA1 512/512 AVX512BW 16x])
Loaded hashes with cost 1 (HMAC size) varying from 6301683 to 2553887820
Will run 32 OpenMP threads
Press 'q' or Ctrl-C to abort, 'h' for help, almost any other key for status
Enabling duplicate candidate password suppressor

Resume is available for the next session...
```

**Comment by illsk1lls** [comment]
Do you have a GPU?  That "Warning" will appear if you are in CPU mode but the file supports GPU mode for faster cracking.. So I'm guessing you do not have a supported GPU otherwise there is a bug in the script.

The bottom line where it says "Enabling duplicate candidate password suppressor" the script is running, you can press a key for status, but once you quit then it should let you know if resume is available or not for the job... is that how that all worked for you?

**Reply by Theletha** [reply]
i have a rtx 4090 with the 7950x3D cpu. Is it maybe trying to run on the iGPU on the cpu? 
As soon as i press a key to continue it just closes the window, but is it actually running cause it didnt feel like it?
![image](https://github.com/user-attachments/assets/b3e0dcae-665a-470f-b5c0-70ca89368d05)

**Reply by Theletha** [reply]
here is how the window looks like when i resume it again after it had quit. If i press a key it will just close the window asap. 

![image](https://github.com/user-attachments/assets/7a3bb90f-91a8-4194-9692-cd879e11daa1)

**Comment by illsk1lls** [comment]
You will probably have to start the job over, but i just updated the script to choose nVidia over AMD if both are present, as that would likely be the scenario for most setups

lmk

**Reply by Theletha** [reply]
hm the output seems the same for the new release, i did choose to start over also
![image](https://github.com/user-attachments/assets/1cb84a12-eba4-443d-aefb-2d07e79b3d6c)

**Reply by illsk1lls** [reply]
Can you post the output of this command entered into CMD prompt?

`WMIC path Win32_VideoController get Name`

**Reply by Theletha** [reply]
> Can you post the output of this command entered into CMD prompt?
> 
> `WMIC path Win32_VideoController get Name`

![image](https://github.com/user-attachments/assets/f250a30e-5a34-4c54-a2a8-2a621db102a8)

**Comment by illsk1lls** [comment]
I made some delayed expansion updates i still haven't been able to reproduce your issue on my side, I'll be adding logging soon.

**Reply by Theletha** [reply]
I see, ill await your next logging update then!

**Reply by illsk1lls** [reply]
I know its been a while but the script is recently updated. It may fix the issue you were having.

**Reply by Theletha** [reply]
> I know its been a while but the script is recently updated. It may fix the issue you were having.

i tried to run the new version but it gives me this info and when i press the any key to continue it just quits. there was prompt earlier asking me if i want to continue the last session but i said no in hopes of it starting from scratch again but it seems to be doing the same thing.

![image](https://github.com/user-attachments/assets/a9829c36-a112-49e3-832f-740130c85b10)

**Comment by illsk1lls** [comment]
Can you check something and add REM and a space to the beggining of line 187

187:`	>nul 2>&1 DEL "%ProgramData%\ignore.Radeon" /F /Q`

187:`REM >nul 2>&1 DEL "%ProgramData%\ignore.Radeon" /F /Q`

Then make an empty file named ignore.Radeon and put it in %ProgramData%

and try running the script

Trying to pinpoint the issue

**Reply by Theletha** [reply]
> Can you check something and add REM and a space to the beggining of line 187
> 
> 187:` >nul 2>&1 DEL "%ProgramData%\ignore.Radeon" /F /Q`
> 
> 187:`REM >nul 2>&1 DEL "%ProgramData%\ignore.Radeon" /F /Q`
> 
> Then make an empty file named ignore.Radeon and put it in %ProgramData%
> 
> and try running the script
> 
> Trying to pinpoint the issue

its doing something now but it kinda gets stuck and i have to manually press a button for it to refresh the line. And it seems its not using any gpu %. The ETA looks a bit crazy too
![image](https://github.com/user-attachments/assets/2debccc7-3899-4136-9420-9022b3cc179f)

**Reply by illsk1lls** [reply]
Thats running on CPU

**Reply by Theletha** [reply]
> Thats running on CPU

this is how it looks now. not sure why it doesnt want to engage gpu i am on all the latest updates and all.

![image](https://github.com/user-attachments/assets/137cf87a-b454-45e1-af7d-6510e9e97e23)

**Comment by illsk1lls** [comment]
If you make a batch script with the below content what is the output? 0, 1, or 2?

This is taken directly from ZipRipper

TestGPUDetection.cmd

```
@ECHO OFF
CALL :CHECKGPU
SETLOCAL ENABLEDELAYEDEXPANSION
ECHO !GPU!
ENDLOCAL
PAUSE
GOTO :EOF

:CHECKGPU
FOR /F "usebackq skip=1 tokens=2,3" %%# IN (`WMIC path Win32_VideoController get Name`) DO (
	IF /I "%%#"=="GeForce" (
		IF /I NOT EXIST "%WinDir%\System32\OpenCL.dll" (
			IF /I EXIST "%WinDir%\System32\DriverStore\FileRepository\nvamig.inf_amd64_72a8482547fd21bc\OpenCL64.dll" (
				>nul 2>&1 COPY /Y "%WinDir%\System32\DriverStore\FileRepository\nvamig.inf_amd64_72a8482547fd21bc\OpenCL64.dll" "%WinDir%\System32\OpenCL.dll"
				SET GPU=1
			) ELSE (
				SET GPU=0
			)
		) ELSE (
			SET GPU=1
		)
	)
	IF /I "%%#"=="RTX" (
		IF /I NOT EXIST "%WinDir%\System32\OpenCL.dll" (
			SET GPU=0
		) ELSE (
			SET GPU=1
		)
	)
	IF /I "%%#"=="A800" (
		IF /I NOT EXIST "%WinDir%\System32\OpenCL.dll" (
			SET GPU=0
		) ELSE (
			SET GPU=1
		)
	)
	IF /I "%%#"=="T1000" (
		IF /I NOT EXIST "%WinDir%\System32\OpenCL.dll" (
			SET GPU=0
		) ELSE (
			SET GPU=1
		)
	)
	IF /I "%%#"=="Quadro" (
		IF /I NOT EXIST "%WinDir%\System32\OpenCL.dll" (
			IF /I EXIST "%WinDir%\System32\DriverStore\FileRepository\nvamig.inf_amd64_72a8482547fd21bc\OpenCL64.dll" (
				>nul 2>&1 COPY /Y "%WinDir%\System32\DriverStore\FileRepository\nvamig.inf_amd64_72a8482547fd21bc\OpenCL64.dll" "%WinDir%\System32\OpenCL.dll"
				SET GPU=1
			) ELSE (
				SET GPU=0
			)
		) ELSE (
			SET GPU=1
		)
	)
	SETLOCAL ENABLEDELAYEDEXPANSION
	IF NOT "!GPU!"=="1" (
		ENDLOCAL
		IF /I NOT EXIST "%ProgramData%\ignore.Radeon" (
			IF /I "%%#"=="Radeon" (
				IF /I NOT EXIST "%WinDir%\System32\amdocl64.dll" (
					CALL :FIXRADEON
				) ELSE (
					REG QUERY "HKEY_LOCAL_MACHINE\SOFTWARE\Khronos\OpenCL\Vendors" /v "%WinDir%\System32\amdocl64.dll" >nul 2>&1
					IF %ERRORLEVEL% NEQ 0 (
						CALL :FIXRADEON
					)
				)
				IF /I "%%# %%$"=="Radeon RX" (
					IF /I NOT EXIST "%WinDir%\System32\amdocl64.dll" (
						SET GPU=0
					) ELSE (
						SET GPU=2
					)
				)
				IF /I "%%# %%$"=="Radeon Pro" (
					IF /I NOT EXIST "%WinDir%\System32\amdocl64.dll" (
						SET GPU=0
					) ELSE (
						SET GPU=2
					)
				)
			)
		)
	) ELSE (
		ENDLOCAL
	)
)
EXIT /b
```

**Reply by Theletha** [reply]
output is 1

![image](https://github.com/user-attachments/assets/e4d5b2cc-a76a-47ef-8e8e-c7587e9d7720)

**Reply by illsk1lls** [reply]
Ok well that part is working correctly, it seems its crashing somewhere during the hashing.. i'll keep looking

**Reply by illsk1lls** [reply]
@Theletha I think it's possible the file you have is a different hash type that I havent seen yet.  I tried adding it blindly on my end after looking through some JtR and zip2john documentation, but I can't find any samples.  

Ironically the format in question is simply "zip", lol

Script is updated lmk

**Reply by Theletha** [reply]
After 15min it looks like this 
![image](https://github.com/user-attachments/assets/b1d38e29-79c0-4279-8519-06ef6ed0c236)

**Reply by illsk1lls** [reply]
Can you do me a favor and while its running look in %programdata%\JtR\run at the file called pwhash and tell me what the first line of that says something like
`filename.zip/test.dat:$zip2$*0*3*0*1d15f05ab282162e63dc`
im looking for the hash identifier between the 2 dollar signs.. also are there any dollar signs in the filename?
in the example above its `zip2`

I really thought I had it.. Its happening during the hash identification im sure of that, when its working right at the top it wont say "Initializing..." it will say "GPU Mode" or "CPU Mode"

**Reply by Theletha** [reply]
the file is 2.5GB i cant open it in notepad or notepad++ due to it being so big. Should i use another program to check it?

**Reply by illsk1lls** [reply]
In cmd prompt
```
cd %programdata%\JtR\run
set /p string=< pwhash
echo %string%
```

**Reply by illsk1lls** [reply]
I am able to reproduce on large files, how large is your original file? (2.5gb original 4.5gb hash in testing on my end)

Edit: Got it working, just have to finish testing and I'll update it here

**Reply by illsk1lls** [reply]
Should be good now. Happy Holidays ;P

**Reply by Theletha** [reply]
> Should be good now. Happy Holidays ![;P](https://static-cdn.jtvnw.net/emoticons/v2/13/default/light/1.0)

looking good!
![image](https://github.com/user-attachments/assets/2c8fe3a8-cc07-49e4-9753-01e3f6824ed5)

**Reply by illsk1lls** [reply, marked as answer]
Just keep in mind larger archives take longer, each guess requires a lot more data...

You could crush a complex password on a small file but it takes a lil longer for bigger.  On a 4090 youre basically going as fast as possible.  Good luck with that file.
